data/trie/patriciaMerkleTrie.go

```diff
@@ -316,3 +365,26 @@ func emptyTrie(root []byte) bool {
 	}
 	return false
 }
+
+// Rollback invalidates the hashes that correspond to the given root hash from the eviction waiting list
+func (tr *patriciaMerkleTrie) Rollback(rootHash []byte) error {
+	_, err := tr.dbEvictionWaitingList.Evict(rootHash)
+	return err
+}
+
+// Prune removes from the database all the old hashes that correspond to the given root hash
+func (tr *patriciaMerkleTrie) Prune(rootHash []byte) error {
+	hashes, err := tr.dbEvictionWaitingList.Evict(rootHash)
```

Comment:
I think you also need to remove the root hash from tr.db ?

Reply:
Among the hashes that are mapped to the rootHash, is also the rootHash. So it is removed from the db.